src/test/ui/consts/const-eval/parse_ints.rs

```diff
@@ -0,0 +1,10 @@
+#![feature(const_int_from_str)]
+
+const _: () = match "-1234".parse::<i32>() {
+    Ok(x) => assert!(x == -1234),
+    Err(_) => panic!(),
+};
+const _TOO_LOW: () = { u64::from_str_radix("12345ABCD", 1); };
+const _TOO_HIGH: () = { u64::from_str_radix("12345ABCD", 37); };
```

Comment:
you need to add `//~ ERROR message goes here` annotations to make the test pass

Please also add a `// check-pass` test that shows successful usages of this in a const item initializer.

Reply:
> Please also add a // check-pass test that shows successful usages of this in a const item initializer.

Or better, add these tests in the libcore test suite. ui tests should only be used when regular unit/integration tests are insufficient.

Reply:
> you need to add //~ ERROR message goes here annotations to make the test pass

The test passes fine already; is there some situation in which it is required.

>successful usages of this in a const item initializer.

I have added a const item initializer, `_TEST_CONST_PARSE`, to the libcore test suite.